src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/Initializers/PerfCounterSdkVersionTelemetryInitializer.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.ApplicationInsights.Extensibility.Implementation;
+using System;
+
+namespace Microsoft.Azure.WebJobs.Logging.ApplicationInsights
+{
+    internal class PerfCounterSdkVersionTelemetryInitializer : ITelemetryInitializer
+    {
+        private const string Prefix = "f_";
+        public void Initialize(ITelemetry telemetry)
+        {
+            if (telemetry == null)
+            {
+                return;
+            }
+            
+            if (telemetry is PerformanceCounterTelemetry)
+            {
+                var internalContext = telemetry.Context != null ? telemetry.Context.GetInternalContext() : null;
```

Comment:
prefer telemetry.Context?.GetInternalContext() which is a bit more thread safe.

Reply:
done.